### PR TITLE
Updating latest image references to quay.io

### PR DESCRIPTION
**Pull request by ashleyvjoy**
# Description
Update latest image references to quay.io (Only CSM 1.12 images should point to quay).
As discussed opening this PR so that teams can qualify the driver/modules for quay.io related testing

# GitHub Issues
List the GitHub issues impacted by this PR:

| GitHub Issue # |
| https://github.com/dell/csm/issues/1435 |

# Checklist:

- [ ] I have performed a self-review of my own code to ensure there are no formatting, vetting, linting, or security issues
- [ ] I have verified that new and existing unit tests pass locally with my changes
- [ ] I have not allowed coverage numbers to degenerate
- [ ] I have maintained at least 90% code coverage
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have maintained backward compatibility

# How Has This Been Tested?

> driver:
    Container ID:  containerd://43d2b58ed99794c863bfbf7b4c033dd83434a6fbd53c2232ecc0ec308baf8b5c
    Image:         quay.io/dell/container-storage-modules/csi-vxflexos:nightly
    Image ID:      quay.io/dell/container-storage-modules/csi-vxflexos@sha256:4048761b1378951e22be907328cf52c898cb456390ae35c412bc02555c8c5a14
    Port:          <none>
    Host Port:     <none>
    Command:
      /csi-vxflexos.sh
    Args:
      --array-config=/vxflexos-config/config
      --driver-config-params=/vxflexos-config-params/driver-config-params.yaml
    State:          Running

> Normal  Pulling    50s   kubelet            Pulling image "quay.io/dell/container-storage-modules/csi-vxflexos:nightly"
  Normal  Scheduled  46s   default-scheduler  Successfully assigned vxflexos/vxflexos-controller-679f46f464-c8qvn to worker-1-rjytuassp7jqc
  Normal  Pulled     43s   kubelet            Successfully pulled image "quay.io/dell/container-storage-modules/csi-vxflexos:nightly" in 7.701s (7.701s including waiting). Image size: 91806997 bytes.
  Normal  Created    43s   kubelet            Created container driver
  Normal  Started    42s   kubelet            Started container driver

### CODE REVIEW

**Review by AronAtDell — APPROVED** [review]
LGTM.

**Review by harishp8889 — APPROVED** [review]

**Review by nitesh3108 — APPROVED** [review]

File: config/samples/storage_v1_csm_powerflex.yaml [hunk 1]
```diff
@@ -55,7 +55,7 @@ spec:
       - name: snapshotter
         image: registry.k8s.io/sig-storage/csi-snapshotter:v8.1.0
       - name: csi-metadata-retriever
-        image: dellemc/csi-metadata-retriever:v1.8.0
+        image: quay.io/dell/container-storage-modulescsi-metadata-retriever:v1.8.0
```
Comment (shefali-malhotra) [inline config/samples/storage_v1_csm_powerflex.yaml:58]: The image path is incorrect. Please correct the same.

Comment (shefali-malhotra) [reply config/samples/storage_v1_csm_powerflex.yaml:58]: @harishp8889 Please address the above issue and verify other links as well.

Comment (mgandharva) [reply config/samples/storage_v1_csm_powerflex.yaml:58]: Update in the following PR: https://github.com/dell/csm-operator/pull/759, Please review

**Pull request merged**